rust/arrow/src/error.rs

```diff
@@ -27,6 +27,7 @@ pub enum ArrowError {
     /// Returned when functionality is not yet available.
     NotYetImplemented(String),
     ExternalError(Box<dyn Error + Send + Sync>),
+    CastError(String),
```

Comment:
Adding a new Arrow Error variant seems reasonable to me -- what do you think @jorgecarleitao  and @nevi-me and @andygrove ?

Reply:
I'm happy with the addition